### ISSUE TITLE
[JENKINS-6822] Don't log exception when unable to find memory mon impl

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#4265]
<p>Don't log the exception here, if you want to keep it, then log it as a DEBUG, so that normal operation on a system that doesn't support this monitor doesn't always puke up this stack trace.</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>jvm 1    | WARN  [Monitoring thread for Free Swap Space started on Mon Jun 21 14:35:06 PDT 2010] h.n.SwapSpaceMonitor$1 - Failed to monitor master for Free Swap Space
jvm 1    | java.io.IOException: No suitable implementation found
jvm 1    | 	at org.jvnet.hudson.MemoryMonitor.obtain(MemoryMonitor.java:60) ~[memory-monitor-1.2.jar:na]
jvm 1    | 	at org.jvnet.hudson.MemoryMonitor.get(MemoryMonitor.java:31) ~[memory-monitor-1.2.jar:na]
jvm 1    | 	at hudson.node_monitors.SwapSpaceMonitor$MonitorTask.call(SwapSpaceMonitor.java:103) ~[hudson-core-1.363.jar:na]
jvm 1    | 	at hudson.node_monitors.SwapSpaceMonitor$MonitorTask.call(SwapSpaceMonitor.java:99) ~[hudson-core-1.363.jar:na]
jvm 1    | 	at hudson.remoting.LocalChannel.call(LocalChannel.java:45) ~[remoting-1.363.jar:na]
jvm 1    | 	at hudson.node_monitors.SwapSpaceMonitor$1.monitor(SwapSpaceMonitor.java:83) ~[hudson-core-1.363.jar:na]
jvm 1    | 	at hudson.node_monitors.SwapSpaceMonitor$1.monitor(SwapSpaceMonitor.java:81) ~[hudson-core-1.363.jar:na]
jvm 1    | 	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:200) ~[hudson-core-1.363.jar:na]
</pre>
</div></div>

---
<details><summary><i>Originally reported by <a title="user57" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=user57">user57</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-6822" target="_blank">Don't log exception when unable to find memory mon impl</a></i></summary>
<i><ul>
<li><b>reported by</b>: user57
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2010-08-02T10:12:53+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 0
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>:</i>
<p>Also note that as per <a href="https://issues.jenkins.io/browse/JENKINS-2194" title="Failed to monitor Free Swap Space" class="issue-link" data-issue-key="JENKINS-2194"><del>JENKINS-2194</del></a>, this exception is reported only once per slave, which I think is a reasonable thing to do.</p>

<p>As I mentioned above, for the platforms that Hudson doesn't support the monitoring, please file a separate ticket so that we can expand the support. </p>
